src/System.Runtime.Extensions/src/System/Runtime/Versioning/FrameworkName.cs

```diff
@@ -62,19 +62,25 @@ public String FullName
             {
                 if (_fullName == null)
                 {
-                    StringBuilder sb = new StringBuilder();
-                    sb.Append(Identifier);
-                    sb.Append(c_componentSeparator);
-                    sb.Append(c_versionKey).Append(c_keyValueSeparator);
-                    sb.Append(c_versionValuePrefix);
-                    sb.Append(Version);
-                    if (!String.IsNullOrEmpty(Profile))
+                    if (string.IsNullOrEmpty(Profile))
                     {
-                        sb.Append(c_componentSeparator);
-                        sb.Append(c_profileKey).Append(c_keyValueSeparator);
-                        sb.Append(Profile);
+                        // Roslyn will optimize this to call string.Concat, avoiding any intermediary allocations
+                        _fullName = Identifier + c_componentSeparator + c_versionKey + c_keyValueSeparator + c_versionValuePrefix + Version;
+                    }
+                    else
+                    {
+                        // A string array is allocated here, but intermediary strings are still avoided
+                        var array = new string[]
```

Comment:
Doing it that way would also cause the code to get faster if we ever decided to add an overload to string.Concat that took more arguments and then teach Roslyn to consume it.